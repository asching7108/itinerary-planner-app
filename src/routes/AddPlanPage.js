import React, { Component } from 'react';
import TripContext from '../context/TripContext';
import PlanForm from '../components/PlanForm/PlanForm';

export default class AddPlanPage extends Component {
	static defaultProps = { match: { params: '' } };

	constructor(props) {
		super(props);
		this.state = { error: null };
	}

	static contextType = TripContext;

	componentDidMount() {
		if (this.context.needToUpdate()) {
			this.context.updateTrip(this.props.match.params.trip_id);
		}
	}

	componentDidUpdate() {
		if (!this.state.error && this.context.error) {
			this.setState({ error: this.context.error });
		}
	}

	handleAddPlanSuccess = plans => {
		this.context.updateTrip(plans[0].trip_id);

		const { history } = this.props;
		history.push(`/trip/${plans[0].trip_id}`);
	}

	handleClickOnCancel = () => {
		const { match: { params }, history } = this.props;
		history.push(`/trip/${params.trip_id}`);
	}
	
	render() {
		const { trip } = this.context;
		const { error } = this.state;

		if (error) {
			return <section><h2>{error}</h2></section>;
		}

		return (
			<section className='AddPlanPage'>
				<h2>Create a plan</h2>
				<PlanForm 
					trip={trip}
					location={this.props.location}
					onAddPlanSuccess={this.handleAddPlanSuccess}
					onClickOnCancel={this.handleClickOnCancel}
				/>
			</section>
		);
	}
}