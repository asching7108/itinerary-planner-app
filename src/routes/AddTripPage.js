import React, { Component } from 'react';
import TripListContext from '../context/TripListContext';
import TripForm from '../components/TripForm/TripForm';

export default class AddTripPage extends Component {
	static contextType = TripListContext;

	handleAddTripSuccess = trip => {
		const { tripList, setTripList } = this.context;
		tripList.push(trip);
		setTripList(tripList);

		const { history } = this.props;
		history.push(`/trip/${trip.id}`);
	}

	handleClickOnCancel = () => {
		this.props.history.goBack();
	}
	
	render() {
		return (
			<section className='AddTripPage'>
				<h2>Create a trip</h2>
				<TripForm 
					location={this.props.location}
					onAddTripSuccess={this.handleAddTripSuccess}
					onClickOnCancel={this.handleClickOnCancel}
				/>
			</section>
		);
	}
}