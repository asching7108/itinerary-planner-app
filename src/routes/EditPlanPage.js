import React, { Component } from 'react';
import TripContext from '../context/TripContext';
import PlanForm from '../components/PlanForm/PlanForm';

export default class EditPlanPage extends Component {
	static defaultProps = { match: { params: '' } };

	constructor(props) {
		super(props);
		this.state = {
			plans: [],
			error: null
		};
	}

	static contextType = TripContext;

	componentDidMount() {
		if (this.context.needToUpdate()) {
			this.context.updateTrip(this.props.match.params.trip_id);
		}
	}

	componentDidUpdate() {
		const { plan_id } = this.props.match.params;
		const { planList, error } = this.context;

		if (!this.state.error && error) {
			this.setState({ error });
		}
		
		else if (!this.state.error && !this.state.plans[0] && planList[0]) {
			if (planList.find(p => p.id === Number(plan_id))) {
				this.setState({
					plans: planList.filter(p => p.id === Number(plan_id))
				});
			}
			else {
				this.setState({ error: `Plan doesn't exist` });
			}
		}
	}

	handleUpdatePlanSuccess = plans => {
		this.context.updateTrip(plans[0].trip_id);

		const { history } = this.props;
		history.push(`/trip/${plans[0].trip_id}/plan/${plans[0].id}`);
	}

	handleClickOnCancel = () => {
		const { match: { params }, history } = this.props;
		history.push(`/trip/${params.trip_id}/plan/${params.plan_id}`);
	}
	
	render() {
		const { trip } = this.context;
		const { plans, error } = this.state;
		
		if (error) {
			return <section><h2>{error}</h2></section>;
		}

		return (
			<section className='EditPlanPage'>
				<h2>Edit plan</h2>
				<PlanForm 
					trip={trip}
					plans={plans}
					location={this.props.location}
					onUpdatePlanSuccess={this.handleUpdatePlanSuccess}
					onClickOnCancel={this.handleClickOnCancel}
				/>
			</section>
		);
	}
}