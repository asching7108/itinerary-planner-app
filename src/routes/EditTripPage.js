import React, { Component } from 'react';
import TripListContext from '../context/TripListContext';
import TripForm from '../components/TripForm/TripForm';

export default class EditTripPage extends Component {
	static contextType = TripListContext;

	handleUpdateTripSuccess = trip => {
		const { tripList, setTripList } = this.context;
		tripList.map((t, i) => {
      if (t.id === trip.id) {
        tripList[i] = trip;
      }
    });
		setTripList(tripList);

    const { history } = this.props;
    history.push(`/trip/${trip.id}`);
	}

	handleClickOnCancel = () => {
		const { history } = this.props;
    history.goBack();
	}
	
	render() {
    const { tripList } = this.context;
    const trip_id = this.props.match.params.trip_id;
    const trip = tripList.find(t => t.id == trip_id);
    return (
			<section className='EditTripPage'>
				<h2>Edit trip</h2>
        <TripForm 
          trip={trip}
          location={this.props.location}
					onUpdateTripSuccess={this.handleUpdateTripSuccess}
					onClickOnCancel={this.handleClickOnCancel}
				/>
			</section>
		);
  }
}