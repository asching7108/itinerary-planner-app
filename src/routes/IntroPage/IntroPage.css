.IntroPage__content-wide {
	display: flex;
	flex-direction: column;
	align-items: center;
	margin-bottom: 300px;
	display: none;
}

.IntroPage__content-narrow {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	text-align: center;
	height: calc(100vh - 115px);
}

.IntroPage__content-wide h1 {
	margin: 50px auto;
}

.IntroPage__content-wide img {
	border: 1px solid #99E2FF;
	border-radius: 3px;
	width: 75%;
	margin: 10px;
	box-shadow: 10px 10px #99E2FF;
	-webkit-transition: all 0.3s;
	-moz-transition: all 0.3s;
	-o-transition: all 0.3s;
	transition: all 0.3s;
	display: none;
}

.IntroPage__content-wide img:hover {
	box-shadow: 15px 15px #99E2FF;
	-webkit-transform: translateX(-3px) translateY(-3px);
	transform: translateX(-3px) translateY(-3px);
}

.IntroPage__content-wide button {
	margin-top: 100px;
}

.IntroPage__content-narrow h2,
.IntroPage__content-narrow button {
	position: relative;
	top: -10vh;
	margin: 15px auto;
	font-size: 1.1em;
}

.IntroPage__btn {
	border-radius: 6px;
	padding: 12px 25px;
	font-size: 1.2em;
	background-color: #99E2FF;
	color: #1C1C1E;
	-webkit-transition: all 0.3s;
	-moz-transition: all 0.3s;
	-o-transition: all 0.3s;
	transition: all 0.3s;
}

.IntroPage__btn:hover {
	-webkit-transform: translateY(-3px);
	transform: translateY(-3px);
	background-color: #FF9E99;
	border: 1px solid #FF9E99;
}

footer {
	text-align: center;
	color: #99E2FF;
	margin-top: -100px;
}

footer p, footer div {
	margin: 15px;
}

footer a {
	-webkit-transition: all 0.5s;
	-moz-transition: all 0.5s;
	-o-transition: all 0.5s;
	transition: all 0.5s;
}

.footer__author {
	font-size: 1em;
}

.footer__author a {
	text-decoration: none;
}

.footer__icon {
	font-size: 1.2em;
	margin: 0 6px;
}

.footer__copyright {
	font-size: 0.8em;
}

footer a:hover {
	cursor: pointer;
	color: #FF9E99;
}

@media screen and (min-width: 360px) {
	.IntroPage__content-narrow {
		height: calc(100vh - 85px);
	}
}

@media screen and (min-width: 450px) {
	.IntroPage__content-narrow {
		height: calc(100vh - 95px);
	}

	.IntroPage__content-narrow h2,
	.IntroPage__content-narrow button {
		font-size: 1.2em;
	}
	
	.footer__author {
		font-size: 1.2em;
	}

	.footer__icon {
		font-size: 1.4em;
	}

	.footer__copyright {
		font-size: 0.9em;
	}
}

@media screen and (min-width: 600px) {
	.IntroPage__content-wide,
	.IntroPage__content-wide img {
		display: inherit;
	}

	.IntroPage__content-narrow {
		display: none;
	}
}