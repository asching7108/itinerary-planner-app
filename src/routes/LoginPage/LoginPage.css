.LoginPage__text {
  margin: 20px auto 10px;
}

.LoginPage__sign-up {
  border: 1px solid #007AFF;
  color: #007AFF;
}