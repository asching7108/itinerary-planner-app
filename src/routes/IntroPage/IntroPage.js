import React, { Component } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import tlp from '../../images/tlp.png';
import tp from '../../images/tp.png';
import pp from '../../images/pp.png';
import TripListContext from '../../context/TripListContext';
import AuthApiService from '../../services/auth-api-service';
import { Button } from '../../components/Utils/Utils';
import './IntroPage.css';

const DEMO_CREDENTIAL = {
	email: 'demo@gmail.com',
	password: 'P@ssw0rd'
}

export default class IntroPage extends Component {
	constructor(props) {
		super(props);
		this.state = { error: false };
	}

	static contextType = TripListContext;
	
	handleClickOnExplore = () => {
		AuthApiService.postLogin({ ...DEMO_CREDENTIAL })
			.then(res => {
				this.context.setAuthState(true);
				this.props.history.push('/');
			})
			.catch(res => {
				this.setState({ error: 'The demo is not available temporarily. Please come back later.' });
			});
	}

	renderExploreBtn() {
		return (
			<>
				<Button type='button' className='IntroPage__btn' onClick={e => this.handleClickOnExplore()}>
					EXPLORE VAMOS
				</Button>
				<div role='alert'>
					{this.state.error && <p className='red'>{this.state.error}</p>}
				</div>
			</>
		);
	}

	renderFooter() {
		return (
			<footer>
				<p className='footer__author'>
					{'Created by '}
					<a href='http://asching7108.github.io/portfolio' target='_blank' rel="noopener noreferrer">
						Esther Lin
					</a>
				</p>
				<div className='IconsDiv'>
					<a href="https://github.com/asching7108/" target="_blank" rel="noopener noreferrer" className='footer__icon'>
						<FontAwesomeIcon icon={['fab', 'github']} />
					</a>
					<a href='http://www.linkedin.com/in/esther-lin-tw/' target='_blank' rel="noopener noreferrer" className='footer__icon'>
						<FontAwesomeIcon icon={['fab', 'linkedin']} />
					</a>
				</div>
				<p className='footer__copyright'>Copyright Â©2020</p>
			</footer>
		)
	}

	render() {
		return (
			<>
				<section className='IntroPage'>
					<div className='IntroPage__content-wide'>
						<h1>Create itineraries can be a breeze.</h1>
						<img src={tlp} alt='Trip list page screenshot' />
						<h1>Easily managed. Smartly displayed.</h1>
						<img src={tp} alt='Trip page screenshot' />
						<h1>Have all the travel infos you need at hand.</h1>
						<img src={pp} alt='Plan page screenshot' />
						{this.renderExploreBtn()}
					</div>
					<div className='IntroPage__content-narrow'>
						<h2>Create, manage and display itineraries like a breeze.</h2>
						{this.renderExploreBtn()}
					</div>
				</section>
				{this.renderFooter()}
			</>
		);
	}
}