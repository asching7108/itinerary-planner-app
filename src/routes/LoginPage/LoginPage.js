import React, { Component } from 'react';
import TripListContext from '../../context/TripListContext';
import LoginForm from '../../components/LoginForm/LoginForm';
import { ButtonLikeLink } from '../../components/Utils/Utils';
import './LoginPage.css';

export default class LoginPage extends Component {
	static contextType = TripListContext;

	static defaultProps = {
		location: {},
		history: {
			push: () => {},
		},
		onAuthStateChange: () => {}
	}

	handleLoginSuccess = () => {
		const { location, history } = this.props;
		const dest = (location.state || {}).from || '/';
		this.context.setAuthState(true);
		history.push(dest);
	}

	handleClickOnCancel = () => {
		this.props.history.goBack();
	}
	
	render() {
		return (
			<section className='LoginPage'>
				<h2>Login</h2>
				<LoginForm
					onLoginSuccess={this.handleLoginSuccess}
					onClickOnCancel={this.handleClickOnCancel}
				/>
				<p className='LoginPage__text'>Doesn't have an account?</p>
				<ButtonLikeLink to='/signup' className='LoginPage__sign-up'>
					Sign up
				</ButtonLikeLink>
			</section>
		);
	}
}