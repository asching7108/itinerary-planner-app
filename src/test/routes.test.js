import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import IntroPage from '../routes/IntroPage/IntroPage';
import LoginPage from '../routes/LoginPage/LoginPage';
import PlanPage from '../routes/PlanPage/PlanPage';
import TripListPage from '../routes/TripListPage/TripListPage';
import TripPage from '../routes/TripPage/TripPage';
import AddPlanPage from '../routes/AddPlanPage';
import AddTripPage from '../routes/AddTripPage';
import EditPlanPage from '../routes/EditPlanPage';
import EditTripPage from '../routes/EditTripPage';
import RegisterPage from '../routes/RegisterPage';
import NotFoundPage from '../routes/NotFoundPage';

describe(`IntroPage Component`, () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<IntroPage />, div);
		ReactDOM.unmountComponentAtNode(div);
	})

	it('renders the UI as expected', () => {
		const wrapper = shallow(<IntroPage />);
		expect(toJson(wrapper)).toMatchSnapshot();
	})
})

describe(`LoginPage Component`, () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<BrowserRouter>
				<LoginPage />
			</BrowserRouter>, 
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	})

	it('renders the UI as expected', () => {
		const wrapper = shallow(<LoginPage />);
		expect(toJson(wrapper)).toMatchSnapshot();
	})
})

describe(`PlanPage Component`, () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<BrowserRouter>
				<PlanPage />
			</BrowserRouter>, 
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	})

	/* Snapshot test to be updated. Context not support with enzyme. */
})

describe(`TripListPage Component`, () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<BrowserRouter>
				<TripListPage />
			</BrowserRouter>, 
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	})

	/* Snapshot test to be updated. Context not support with enzyme. */
})

describe(`TripPage Component`, () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<BrowserRouter>
				<TripPage />
			</BrowserRouter>, 
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	})

	/* Snapshot test to be updated. Context not support with enzyme. */
})

describe(`AddPlanPage Component`, () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<AddPlanPage />, div);
		ReactDOM.unmountComponentAtNode(div);
	})

	/* Snapshot test to be updated. Context not support with enzyme. */
})

describe(`AddTripPage Component`, () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<AddTripPage />, div);
		ReactDOM.unmountComponentAtNode(div);
	})

	it('renders the UI as expected', () => {
		const wrapper = shallow(<AddTripPage />);
		expect(toJson(wrapper)).toMatchSnapshot();
	})
})

describe(`EditPlanPage Component`, () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<EditPlanPage />, div);
		ReactDOM.unmountComponentAtNode(div);
	})

	/* Snapshot test to be updated. Context not support with enzyme. */
})

describe(`EditTripPage Component`, () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<EditTripPage />, div);
		ReactDOM.unmountComponentAtNode(div);
	})

	/* Snapshot test to be updated. Context not support with enzyme. */
})

describe(`NotFoundPage Component`, () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<NotFoundPage />, div);
		ReactDOM.unmountComponentAtNode(div);
	})

	it('renders the UI as expected', () => {
		const wrapper = shallow(<NotFoundPage />);
		expect(toJson(wrapper)).toMatchSnapshot();
	})
})

describe(`RegisterPage Component`, () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<RegisterPage />, div);
		ReactDOM.unmountComponentAtNode(div);
	})

	it('renders the UI as expected', () => {
		const wrapper = shallow(<RegisterPage />);
		expect(toJson(wrapper)).toMatchSnapshot();
	})
})