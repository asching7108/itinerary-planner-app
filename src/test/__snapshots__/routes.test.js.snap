// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTripPage Component renders the UI as expected 1`] = `
<section
  className="AddTripPage"
>
  <h2>
    Create a trip
  </h2>
  <TripForm
    location={
      Object {
        "pathname": "",
      }
    }
    onAddTripSuccess={[Function]}
    onClickOnCancel={[Function]}
    trip={Object {}}
  />
</section>
`;

exports[`IntroPage Component renders the UI as expected 1`] = `
<Fragment>
  <section
    className="IntroPage"
  >
    <div
      className="IntroPage__content-wide"
    >
      <h1>
        Create itineraries can be a breeze.
      </h1>
      <img
        alt="Trip list page screenshot"
        src="tlp.png"
      />
      <h1>
        Easily managed. Smartly displayed.
      </h1>
      <img
        alt="Trip page screenshot"
        src="tp.png"
      />
      <h1>
        Have all the travel infos you need at hand.
      </h1>
      <img
        alt="Plan page screenshot"
        src="pp.png"
      />
      <Button
        className="IntroPage__btn"
        onClick={[Function]}
        type="button"
      >
        EXPLORE VAMOS
      </Button>
      <div
        role="alert"
      />
    </div>
    <div
      className="IntroPage__content-narrow"
    >
      <h2>
        Create, manage and display itineraries like a breeze.
      </h2>
      <Button
        className="IntroPage__btn"
        onClick={[Function]}
        type="button"
      >
        EXPLORE VAMOS
      </Button>
      <div
        role="alert"
      />
    </div>
  </section>
  <footer>
    <p
      className="footer__author"
    >
      Created by 
      <a
        href="http://asching7108.github.io/portfolio"
        rel="noopener noreferrer"
        target="_blank"
      >
        Esther Lin
      </a>
    </p>
    <div
      className="IconsDiv"
    >
      <a
        className="footer__icon"
        href="https://github.com/asching7108/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fab",
              "github",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </a>
      <a
        className="footer__icon"
        href="http://www.linkedin.com/in/esther-lin-tw/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fab",
              "linkedin",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </a>
    </div>
    <p
      className="footer__copyright"
    >
      Copyright Â©2020
    </p>
  </footer>
</Fragment>
`;

exports[`LoginPage Component renders the UI as expected 1`] = `
<section
  className="LoginPage"
>
  <h2>
    Login
  </h2>
  <LoginForm
    onClickOnCancel={[Function]}
    onLoginSuccess={[Function]}
  />
  <p
    className="LoginPage__text"
  >
    Doesn't have an account?
  </p>
  <LinkButton
    className="LoginPage__sign-up"
    to="/signup"
  >
    Sign up
  </LinkButton>
</section>
`;

exports[`NotFoundPage Component renders the UI as expected 1`] = `
<section
  className="NotFoundPage"
>
  <h2>
    404 - Page not found
  </h2>
</section>
`;

exports[`RegisterPage Component renders the UI as expected 1`] = `
<section
  className="RegisterPage"
>
  <h2>
    Create an acount
  </h2>
  <RegisterForm
    onClickOnCancel={[Function]}
    onRegisterSuccess={[Function]}
  />
</section>
`;
